src/middleware/Middleware.sol

```diff
@@ -402,24 +411,45 @@ contract Middleware is SimpleKeyRegistry32, Ownable {
     }
 
     /**
-     * @dev Calculates total stake for an epoch
-     * @param epoch The epoch to calculate stake for
-     * @return totalStake The total stake amount
+     * @notice Gets how many operators were active at a specific epoch
+     * @param epoch The epoch at which to check how many operators were active
+     * @return activeOperators The array of active operators
      */
-    function _calcTotalStake(
+    function getCurrentOperators(
         uint48 epoch
-    ) private view returns (uint256 totalStake) {
+    ) external view returns (address[] memory activeOperators) {
         uint48 epochStartTs = getEpochStartTs(epoch);
 
-        // for epoch older than i_slashingWindow total stake can be invalidated (use cache)
-        if (epochStartTs < Time.timestamp() - i_slashingWindow) {
-            revert Middleware__TooOldEpoch();
+        activeOperators = new address[](s_operators.length());
+        uint256 valIdx = 0;
+        for (uint256 i; i < s_operators.length(); ++i) {
+            (address operator, uint48 enabledTime, uint48 disabledTime) = s_operators.atWithTimes(i);
+
+            // just skip operator if it was added after the target epoch or paused
+            if (!_wasActiveAt(enabledTime, disabledTime, epochStartTs)) {
+                continue;
+            }
+
+            activeOperators[valIdx++] = operator;
         }
 
-        if (epochStartTs > Time.timestamp()) {
-            revert Middleware__InvalidEpoch();
+        assembly {
```

Comment:
why through mstore

Reply:
It's to shrink down the array and avoid wasting memory. It's how symbiotic does it too.